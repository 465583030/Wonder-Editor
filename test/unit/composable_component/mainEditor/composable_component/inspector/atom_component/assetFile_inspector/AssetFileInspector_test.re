/* open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe(
    "AssetFileInspector",
    () => {
      let sandbox = getSandboxDefaultVal();
      let _getFromArray = (array, index) => ArrayService.getNth(index, array);
      beforeEach(
        () => {
          sandbox := createSandbox();
          MainEditorSceneTool.initStateAndGl(~sandbox, ());
          MainEditorSceneTool.createDefaultScene(
            sandbox,
            MainEditorAssetTool.initAssetTree(MainEditorAssetTool.buildTwoLayerAssetTree)
          );
          MainEditorAssetTool.setFolder2ToBeCurrentAssetChildrenNodeParent()
        }
      );
      afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
      describe(
        "prepare currentSource",
        () => {
          beforeEach(
            () =>
              CurrentSourceEditorService.setCurrentSource(EditorType.AssetFile)
              |> StateLogicService.getAndSetEditorState
          );
          describe(
            "test component snapshot",
            () => {
              test(
                "if hasn't currentAssetTreeNode, show nothing",
                () =>
                  BuildComponentTool.buildInspectorComponent(
                    TestTool.buildEmptyAppState(),
                    InspectorTool.buildFakeAllShowComponentConfig()
                  )
                  |> ReactTestTool.createSnapshotAndMatch
              );
              describe(
                "else",
                () => {
                  test(
                    "test show image file",
                    () => {
                      MainEditorAssetTool.setImgFileToBeCurrentAssetTreeNode();
                      BuildComponentTool.buildInspectorComponent(
                        TestTool.buildEmptyAppState(),
                        InspectorTool.buildFakeAllShowComponentConfig()
                      )
                      |> ReactTestTool.createSnapshotAndMatch
                    }
                  );
                  test(
                    "test show json file",
                    () => {
                      MainEditorAssetTool.setJsonFileToBeCurrentAssetTreeNode();
                      BuildComponentTool.buildInspectorComponent(
                        TestTool.buildEmptyAppState(),
                        InspectorTool.buildFakeAllShowComponentConfig()
                      )
                      |> ReactTestTool.createSnapshotAndMatch
                    }
                  )
                }
              )
            }
          );
          describe(
            "test rename file",
            () => {
              let triggerChangeEvent = (value, domChildren) => {
                let article = _getFromArray(domChildren, 0);
                let input = _getFromArray(article##children, 3);
                BaseEventTool.triggerChangeEvent(input, BaseEventTool.buildFormEvent(value))
              };
              let triggerBlurEvent = (value, domChildren) => {
                let article = _getFromArray(domChildren, 0);
                let input = _getFromArray(article##children, 3);
                BaseEventTool.triggerBlurEvent(input, BaseEventTool.buildFormEvent(value))
              };
              test(
                "test not rename",
                () =>
                  BuildComponentTool.buildAssetFileContentComponent()
                  |> ReactTestTool.createSnapshotAndMatch
              );
              test(
                "test rename to specific name",
                () => {
                  MainEditorAssetTool.setImgFileToBeCurrentAssetTreeNode();
                  let newName = "arvin";
                  let component =
                    BuildComponentTool.buildAssetFileInspector(
                      MainEditorAssetTool.imgFileId,
                      MainEditorAssetTool.buildFakeImgFileResult()
                    );
                  BaseEventTool.triggerComponentEvent(component, triggerChangeEvent(newName));
                  BaseEventTool.triggerComponentEvent(component, triggerBlurEvent(newName));
                  BuildComponentTool.buildAssetFileContentComponent()
                  |> ReactTestTool.createSnapshotAndMatch
                }
              );
              describe(
                "deal with specific case",
                () =>
                  test(
                    "key in '', trigger onBlur, the input value should be origin name",
                    () => {
                      MainEditorAssetTool.setImgFileToBeCurrentAssetTreeNode();
                      let newName = "";
                      let component =
                        BuildComponentTool.buildAssetFileInspector(
                          MainEditorAssetTool.imgFileId,
                          MainEditorAssetTool.buildFakeImgFileResult()
                        );
                      BaseEventTool.triggerComponentEvent(component, triggerChangeEvent(newName));
                      BaseEventTool.triggerComponentEvent(component, triggerBlurEvent(newName));
                      component |> ReactTestTool.createSnapshotAndMatch
                    }
                  )
              )
            }
          )
        }
      )
    }
  ); */