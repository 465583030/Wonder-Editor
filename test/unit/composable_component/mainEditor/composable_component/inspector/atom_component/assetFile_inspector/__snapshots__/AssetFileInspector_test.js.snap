// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetFileInspector test show fileContent else test show image file 1`] = `
<article
  className="inspector-component"
>
  <article
    className="inspector-component"
  >
    <div
      className=""
    >
      <h1>
        Image
      </h1>
      <hr />
      <span
        className=""
      >
        name:
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="1.jpg"
      />
    </div>
  </article>
</article>
`;

exports[`AssetFileInspector test show fileContent else test show json file 1`] = `
<article
  className="inspector-component"
>
  <article
    className="inspector-component"
  >
    <div>
      <h1>
        Json
      </h1>
      <hr />
      <span
        className=""
      >
        name:
      </span>
      <input
        className="input-component float-input"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="2.json"
      />
      <p>
        2.json
      </p>
    </div>
  </article>
</article>
`;

exports[`AssetFileInspector test show fileContent if hasn't currentFile, show nothing 1`] = `
<article
  className="inspector-component"
/>
`;
